### PR TITLE
Add number of shots statistics

### PR DESCRIPTION
**Pull request by MichalKalke**
Add number of shots of the whole game for each team to the statistics table

### CODE REVIEW

**Review by marekkawalski — APPROVED** [review]

File: frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import './GameStatistics.css';
+
+function NumberOfShots(statistics) {
+  return (
+    <>
+      <div className="table-item">{statistics.blue}</div>
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js:7]: Are you sure that `statistics` will exist every time?

File: frontend/smart-kickers-game/src/components/Game/GameStatistics/GameStatistics.js [hunk 2]
```diff
@@ -65,6 +66,7 @@ function GameStatistics({ finalScores, onNewGameRequested }) {
         <div className="table-item">{getManualSubstractedGoals(TeamID.Team_blue)}</div>
         <div className="table-item">Manually substracted goals</div>
         <div className="table-item">{getManualSubstractedGoals(TeamID.Team_white)}</div>
+        <NumberOfShots blue={statistics.blueShotsCount} white={statistics.whiteShotsCount} />
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/components/Game/GameStatistics/GameStatistics.js:69]: `NumberOfShots` component gets `statistics`, not `blue` and `white`

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
As in comments and one question, have you tested that IRL?

File: frontend/smart-kickers-game/src/components/Game/GameStatistics/GameStatistics.js (hunk 2 above)
Comment (marekkawalski) [reply frontend/smart-kickers-game/src/components/Game/GameStatistics/GameStatistics.js:69]: It has been fixed

File: frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js (hunk 1 above)
Comment (marekkawalski) [reply frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js:7]: It has been fixed. It doesn't exist every time.

File: frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import './GameStatistics.css';
+
+function NumberOfShots({ statistics }) {
+  console.log(statistics);
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js]: image

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
What if `statistics` will be undefined?

File: frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js (hunk 3 above)
Comment (marekkawalski) [reply frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js]: We are now protected against undefined statistics
file

File: frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js [hunk 4]
```diff
@@ -4,9 +4,9 @@ import './GameStatistics.css';
 function NumberOfShots({ statistics }) {
   return (
     <>
-      <div className="table-item">{statistics?.blueShotsCount}</div>
+      <div className="table-item">{statistics ? statistics.blueShotsCount : '0'}</div>
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js:7]: In theory `statistics` should be here every time now, but `blueShotsCount` probably not.

File: frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js [hunk 5]
```diff
@@ -4,9 +4,9 @@ import './GameStatistics.css';
...
       <div className="table-item">number of all shots in the game</div>
-      <div className="table-item">{statistics?.whiteShotsCount}</div>
+      <div className="table-item">{statistics ? statistics.blueShotsCount : '0'}</div>
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js:9]: Second time `blueShotsCount`?

File: frontend/smart-kickers-game/src/components/Game/GameStatistics/GameStatistics.js [hunk 6]
```diff
@@ -66,7 +66,7 @@ function GameStatistics({ finalScores, onNewGameRequested }) {
         <div className="table-item">{getManualSubstractedGoals(TeamID.Team_blue)}</div>
         <div className="table-item">Manually substracted goals</div>
         <div className="table-item">{getManualSubstractedGoals(TeamID.Team_white)}</div>
-        <NumberOfShots statistics={statistics} />
+        {statistics && <NumberOfShots statistics={statistics} />}
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/components/Game/GameStatistics/GameStatistics.js:69]: Now you aren't displaying `NumberOfShots` when statistics aren't provided, maybe better option is to add
```
if(statistics) return <p>Something go wrong, statistics goes on the vacation and we don't have it</p>
```
Just an option and sugestion.

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
image

**Comment by mrCherry97** [comment]
And in the first step resolve conflicts. :p
image

File: frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js (hunk 4 above)
Comment (MichalKalke) [reply frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js]: Changed to statistics?.blueShotsCount and same for whiteShotsCount.

File: frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js (hunk 5 above)
Comment (MichalKalke) [reply frontend/smart-kickers-game/src/components/Game/GameStatistics/NumberOfShots.js]: Replaced second to `whiteShotsCount`

File: frontend/smart-kickers-game/src/components/Game/GameStatistics/GameStatistics.js (hunk 6 above)
Comment (MichalKalke) [reply frontend/smart-kickers-game/src/components/Game/GameStatistics/GameStatistics.js]: Thanks for that, we used it.

**Review by mrCherry97 — COMMENTED** [review]
I hope it works.
I will make `if` in `NumberOfShots` instead of `{statistics?.fastestShot.speed !== 0 && statistics ?` but I think this is fine, your decision. 
Resolve conflicts and LGTM

**Pull request merged**